satpy/dependency_tree.py

```diff
@@ -488,14 +496,43 @@ def _promote_query_to_modified_dataid(self, query, dep_key):
                 orig_dict[key] = dep_val
         return dep_key.from_dict(orig_dict)
 
-    def get_compositor(self, key):
+    def get_compositor(self, key: DataQuery):
         """Get a compositor."""
         for sensor_name in sorted(self.compositors):
             try:
                 return self.compositors[sensor_name][key]
             except KeyError:
                 continue
 
+        if key.get("name", default="*") != "*" and len(key.to_dict()) == 1:
+            # the query key is just the name and still couldn't be found
+            raise KeyError("Could not find compositor '{}'".format(key))
+
+        # Get the generic version of the compositor (by name only)
+        # then save our new version under the new name
+
+        return self._get_compositor_by_name(key)
```

Comment:
What use cases in this covering?

Reply:
Ah the name of this method is incorrect after more refactoring was done. So the code above this is when the compositor exactly matches what the user asked for. This is usually the common case of just a compositor by name but could also include other filtering parameters like `resolution` or `calibration` or whatever else was asked for.

The method `_get_compositor_by_name` first gets the base compositor by just the name and then if more than one compositor matches it filters down using the other properties of the provided `DataQuery`. The cases where a compositor name returns more than one compositor would be if there are varying resolutions or calibrations for a single compositor. So again, not a common use case, but it was in the tests already.